### PR TITLE
Search: Move customizer integration into package

### LINKED ISSUE #23153: S1.1 Migrate custom controls including the class and assets
It allows user to change search settings from customizer, and needs to be migrated.

Be aware the code is still synced with WPCOM and needs to be tested on WPCOM before merging. 

https://github.com/Automattic/jetpack/tree/master/projects/plugins/jetpack/modules/search/customize-controls

### PR DESCRIPTION
**Pull request by jsnmoon**
Fixes #23153.

#### Changes proposed in this Pull Request:

- Moves Instant Search integration code for the WordPress Customizer into the Search package.
- Removes now obsolete references to the `modules/search` folder in the Jetpack plugin.

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Apply this change to your site with a Search subscription.
* Navigate to the WordPress Customizer.
* Ensure that the Jetpack Search section still exists and remains usable.

### CODE REVIEW

File: projects/packages/search/src/customizer/class-customizer.php
Comment (kangzj) [inline projects/packages/search/src/customizer/class-customizer.php:31]: I suspect the two `require_once` are not needed anymore 😄

**Review by kangzj — DISMISSED** [review]
It works as described and changes look awesome. Left only a small comment. Feel free to pass on to crew.

File: projects/packages/search/src/customizer/class-customizer.php
Comment (jsnmoon) [reply projects/packages/search/src/customizer/class-customizer.php:31]: Good catch!

**Review by jeherve — DISMISSED** [review]
This looks good to me. I had one question, but I realize now that it was a stupid one. :) This should be good to merge after a last master merge.

File: projects/packages/search/src/customizer/customize-controls/class-label-control.php
Comment (jsnmoon) [reply projects/packages/search/src/customizer/customize-controls/class-label-control.php:16]: Forgot to note that in the original post! Thanks for the review 🤝

**Comment by jsnmoon** [comment]
@jeherve: Could I get one more 👍? 🙏

**Review by jeherve — APPROVED** [review]

**Comment by jsnmoon** [comment]
D75892-code has shipped!

**Pull request merged**
